File: lib/features/stores/domain/entities/store.dart
class Store {
  final String id;
  final String name;
  final String? description;
  final String? coverImageUrl;
  final String? logoUrl;
  final String category;
  final double rating;
  final int reviewCount;
  final double latitude;
  final double longitude;
  final String address;
  final String? phone;
  final String? email;
  final String? website;
  final bool isOpen;
  final String? openingHours;
  final String? closingHours;
  final double distance;
  final int activeOfferCount;
  final bool isFavorite;
  final List<String> acceptedPaymentMethods;

  const Store({
    required this.id,
    required this.name,
    this.description,
    this.coverImageUrl,
    this.logoUrl,
    required this.category,
    this.rating = 0,
    this.reviewCount = 0,
    required this.latitude,
    required this.longitude,
    required this.address,
    this.phone,
    this.email,
    this.website,
    this.isOpen = false,
    this.openingHours,
    this.closingHours,
    this.distance = 0,
    this.activeOfferCount = 0,
    this.isFavorite = false,
    this.acceptedPaymentMethods = const [],
  });
}

File: backend/src/LastHour.Domain/Entities/Store.cs
using LastHour.Domain.Common;
using LastHour.Domain.Exceptions;

namespace LastHour.Domain.Entities;

public class Store : BaseAuditableEntity
{
    public string OwnerId { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? CoverImageUrl { get; set; }
    public string? LogoUrl { get; set; }
    public string Category { get; set; } = string.Empty;
    public double Rating { get; set; }
    public int ReviewCount { get; set; }
    public double Latitude { get; set; }
    public double Longitude { get; set; }
    public string Address { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public string? Email { get; set; }
    public string? Website { get; set; }
    public bool IsOpen { get; set; }
    public string? OpeningHours { get; set; }
    public string? ClosingHours { get; set; }
    public int ActiveOfferCount { get; set; }
    public bool IsActive { get; set; } = true;
    public string? AcceptedPaymentMethods { get; set; }

    public ICollection<Offer> Offers { get; set; } = [];
    public ICollection<Order> Orders { get; set; } = [];
    public ICollection<Review> Reviews { get; set; } = [];
    public ICollection<Favorite> Favorites { get; set; } = [];
}
